### PR TITLE
Prevent user overriding with admin:loadSecurities

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

Actually when users are loaded with `admin:loadSecurities`, they are first deleted if they already exist.  
So if those users had API key, they will be deleted as well.

The behavior can be confusing so by default user overriding is now forbidden.  
The argument `onExistingUsers` can be set at `skip` or `override` to skip or override existing users

### How should this be manually tested?

```
$ NODE_ENV=production docker-compose up -d
$ kourou query admin:loadSecurities --body '{ users: { admin: {content: {profileIds:["admin"] }}}}'
$ kourou query admin:loadSecurities --body '{ users: { admin: {content: {profileIds:["admin"] }}}}'
```

### CODE REVIEW

**Comment by scottinet** [comment]
I tend to dislike behavior changes in the API between different environments, as that usually means added complexity to maintenance and support (env variables are often overlooked when trying to reproduce problem).

I wonder if the behavior shouldn't be changed to something like:
* (default) fail if loaded security objects already exist
* (optional) instead, silently skip already existing objects
* (optional) instead, replace existing objects with the ones loaded

That way, we have several options that may be of use depending of the use case (I think I can imagine a few use cases that would benefit for either ones of these behaviors), while having an explicitly and so perfectly predictible behavior.

What do you think?

**Comment by Aschen** [comment]
I totally agree :+1: 
IMHO it's the responsibility of the client to use env variable to change the application behavior.

I will modify the option accordingly.

**Comment by scottinet** [comment]
@Aschen 
> IMHO it's the responsibility of the client to use env variable to change the application behavior.

Then we must have a documentation page, with the exhaustive list of changes between development and production environments.
Until we have such a page, users cannot have the complete information they need to properly decide what value to give to their NODE_ENV variable and what are the impacts.

**Comment by Aschen** [comment]
> @Aschen
> 
> > IMHO it's the responsibility of the client to use env variable to change the application behavior.
> 
> Then we must have a documentation page, with the exhaustive list of changes between development and production environments.
> Until we have such a page, users cannot have the complete information they need to properly decide what value to give to their NODE_ENV variable and what are the impacts.

Yes I agree, what I wanted to say is that the user will use `NODE_ENV` or whatever to change the payload he is sending to Kuzzle.
Kuzzle will not act differently according to environment variables

File: doc/2/api/controllers/admin/load-securities/index.md [hunk 1]
```diff
@@ -16,6 +16,13 @@ The roles, profiles and users definitions follow the same structure as in the bo
  - [createProfile](/core/2/api/controllers/security/create-profile)
  - [createUser](/core/2/api/controllers/security/create-user)
 
+If some users already exists, they will be deleted and then created again.
+
+::: warning
+Kuzzle prevent existing user overriding in production environment.
```
[the remark this answers is not in the document]
Comment (scottinet) [reply doc/2/api/controllers/admin/load-securities/index.md:22]: From the code it seems that there is no difference between dev and production environments. If so you should modify this sentence to reflect that.

File: lib/api/controllers/admin.js [hunk 2]
```diff
@@ -21,13 +21,30 @@
...
+ * @returns {String}
+ */
+function getWaitForRefresh (request) {
+  if ( request.input.args.refresh === false
+    || request.input.args.refresh === 'false'
+  ) {
+    return false;
+  }
+
+  return true;
```
[the remark this answers is not in the document]
Comment (Aschen) [reply lib/api/controllers/admin.js:45]: For undefined yes but not for `null`. I updated the method in `BaseController`

File: doc/2/api/controllers/admin/load-securities/index.md [hunk 3]
```diff
@@ -16,6 +16,13 @@ The roles, profiles and users definitions follow the same structure as in the bo
  - [createProfile](/core/2/api/controllers/security/create-profile)
  - [createUser](/core/2/api/controllers/security/create-user)
 
+If users already exist, they will be deleted and then created again.
+
+::: warning
+By default, Kuzzle prevent existing user overwriting.
```
Comment (Leodau) [inline doc/2/api/controllers/admin/load-securities/index.md:22]:
```suggestion
By default, Kuzzle prevents existing user overwriting.  
```

**Pull request merged**